tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/cassandra/tpch/CassandraTpchDataSource.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.tempto.internal.fulfillment.table.cassandra.tpch;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.teradata.tempto.fulfillment.table.jdbc.RelationalDataSource;
+import com.teradata.tempto.internal.query.QueryRowMapper;
+import io.airlift.tpch.TpchEntity;
+import io.airlift.tpch.TpchTable;
+
+import java.sql.JDBCType;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class CassandraTpchDataSource
+    implements RelationalDataSource
+{
+    private static final Splitter SPLITTER = Splitter.on('|');
+
+    private final TpchTable table;
+    private final List<Integer> mapping;
+    private final List<JDBCType> columnTypes;
+    private final double scaleFactor;
+
+    public CassandraTpchDataSource(TpchTable table, List<Integer> mapping, List<JDBCType> columns, double scaleFactor)
```

Comment:
What does this look like from a user perspective? You pass in a list of integers that specify the order that you want the tpch data to come out? I think it would be clearer from a user perspective for them to specify only the primary key(s) and handle the sorting here.